network/test/unicast_authorization_test.go

```diff
@@ -408,3 +399,60 @@ func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_PublicChannel()
 	// wait for slashing violations consumer mock to invoke run func and close ch if expected method call happens
 	unittest.RequireCloseBefore(u.T(), u.waitCh, u.channelCloseDuration, "could close ch on time")
 }
+
+// TestUnicastAuthorization_UnauthorizedUnicastOnChannel tests that messages sent via unicast that are not authorized for unicast are rejected.
+func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_UnauthorizedUnicastOnChannel() {
+	// setup mock slashing violations consumer and middlewares
+	slashingViolationsConsumer := mocknetwork.NewViolationsConsumer(u.T())
+	u.setupMiddlewaresAndProviders(slashingViolationsConsumer)
+
+	// set sender id role to RoleConsensus to avoid unauthorized sender validation error
+	u.senderID.Role = flow.RoleConsensus
+
+	expectedSenderPeerID, err := unittest.PeerIDFromFlowID(u.senderID)
+	require.NoError(u.T(), err)
+
+	expectedViolation := &slashing.Violation{
+		Identity:  u.senderID,
+		PeerID:    expectedSenderPeerID.String(),
+		MsgType:   "BlockProposal",
+		Channel:   channels.ConsensusCommittee,
+		IsUnicast: true,
+		Err:       message.ErrUnauthorizedUnicastOnChannel,
+	}
+
+	slashingViolationsConsumer.On(
+		"OnUnauthorizedUnicastOnChannel",
+		expectedViolation,
+	).Once().Run(func(args mockery.Arguments) {
+		close(u.waitCh)
+	})
+
+	overlay := &mocknetwork.Overlay{}
```

Comment:
```suggestion
	overlay := mocknetwork.NewOverlay(u.T())
```
Please also refactor this elsewhere in the code, when using `testify` constructor, it also asserts expectations.

Reply:
https://github.com/onflow/flow-go/pull/3378/commits/29caa339e0fe988e6e4cd5076a026056a531e14a